dpnp/tests/test_statistics.py

```diff
@@ -127,6 +128,186 @@ def test_avg_error(self):
+                    # also assert_dtype_allclose for integer types is
+                    # always check for exact match
+                    rdtype = dpnp.default_float_type(ad.device)
+
+                result = result.astype(rdtype)
+
+                if rdtype == dpnp.bool:
+                    result = result.astype(dpnp.int32)
+                    rdtype = result.dtype
+
+                expected = expected.astype(rdtype)
+
+                factor = 1000
+                rtol = atol = factor_to_tol(rdtype, factor)
+                invalid = numpy.logical_not(
+                    numpy.isclose(
+                        result.asnumpy(), expected, rtol=rtol, atol=atol
+                    )
+                )
+
+                # When using the 'fft' method, we might encounter outliers.
+                # This usually happens when the resulting array contains values close to zero.
+                # For these outliers, the relative error can be significant.
+                # We can tolerate a few such outliers.
+                # max_outliers = 10 if expected.size > 1 else 0
+                max_outliers = 10 if expected.size > 1 else 0
+                if invalid.sum() > max_outliers:
+                    # we already failed check,
+                    # call assert_dtype_allclose just to report error nicely
+                    assert_dtype_allclose(result, expected, factor=factor)
+
+    def test_convolve_mode_error(self):
+        a = dpnp.arange(5)
+        v = dpnp.arange(3)
+
+        # invalid mode
+        with pytest.raises(ValueError):
+            dpnp.convolve(a, v, mode="unknown")
+
+    @pytest.mark.parametrize("a, v", [([], [1]), ([1], []), ([], [])])
+    def test_convolve_empty(self, a, v):
+        a = dpnp.asarray(a)
+        v = dpnp.asarray(v)
+
+        with pytest.raises(ValueError):
+            dpnp.convolve(a, v)
+
+    @pytest.mark.parametrize("a, v", [([1], 2), (3, [4]), (5, 6)])
+    def test_convolve_scalar(self, a, v):
+        an = numpy.asarray(a, dtype=numpy.float32)
+        vn = numpy.asarray(v, dtype=numpy.float32)
+
+        ad = dpnp.asarray(a, dtype=numpy.float32)
+        vd = dpnp.asarray(v, dtype=numpy.float32)
+
+        expected = numpy.convolve(an, vn)
+        result = dpnp.convolve(ad, vd)
+
+        assert_dtype_allclose(result, expected)
+
+    @pytest.mark.parametrize(
+        "a, v",
+        [
+            ([[1, 2], [2, 3]], [1]),
+            ([1], [[1, 2], [2, 3]]),
+            ([[1, 2], [2, 3]], [[1, 2], [2, 3]]),
+        ],
+    )
+    def test_convolve_shape_error(self, a, v):
+        a = dpnp.asarray(a)
+        v = dpnp.asarray(v)
+
+        with pytest.raises(ValueError):
+            dpnp.convolve(a, v)
+
+    @pytest.mark.parametrize("size", [2, 10**1, 10**2, 10**3, 10**4, 10**5])
+    def test_convolve_different_sizes(self, size):
+        a = numpy.random.rand(size).astype(numpy.float32)
+        v = numpy.random.rand(size // 2).astype(numpy.float32)
```

Comment:
looks the same as
```suggestion
        a = generate_random_numpy_array(size, dtype=numpy.float32, low=0, high=1)
        v = generate_random_numpy_array(size // 2, dtype=numpy.float32, low=0, high=1)
```
if so, we can remove `setup_method` above.